src/WebInterface.hs

```diff
@@ -240,15 +240,13 @@ viewPullRequestWithApproval info prId pullRequest = do
   case Project.approval pullRequest of
     Just Approval { approver = Username username, approvedFor = approvalType } ->
       span ! class_ "review" $ do
-        let approvedAction = case approvalType of
-              Merge -> "merge"
-              MergeAndDeploy -> "merge and deploy"
-        toHtml $ format "Approved for {} by " [approvedAction :: String]
+        let approvedAction = Project.displayApproval approvalType
+        toHtml $ format "Approved for {} by " [approvedAction]
         -- TODO: Link to approval comment, not just username.
         let url = Text.append "https://github.com/" username
         a ! href (toValue url) $ toHtml username
     Nothing ->
-      fail $
+      error $
```

Comment:
I suppose this is `MonadFail` (or the absence thereof for the `Html` monad) related?